src/components/CardDetails.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+export default class CardDetails extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentPage: props.location.pathname,
+      data: {},
+      loader: true
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.location.pathname === this.props.location.pathname) {
+      return;
+    }
+    this.getJson();
+  }
+
+  componentDidMount() {
+    this.getJson();
+  }
+
+  getJson() {
+    const {currentPage} = this.state;
+    fetch(`https://swapi.co/api${currentPage}/?format=json`)
+    .then(resolve => resolve.json())
+    .then(data => this.setState((prevState) => {
+      return {
+        data: data,
+        loader: false
+      }
+    }));
```

Comment:
will better write as: 
.then(data => {
this.setState(() => ({
        data,
        loader: false
      });
}